From 3951e02ed89b5f2a8c1217e99d05d7b12e842d2f Mon Sep 17 00:00:00 2001
From: bakkeby <bakkeby@gmail.com>
Date: Mon, 1 Jul 2019 10:33:12 +0200
Subject: [PATCH] Lose fullscreen on focus change

---
 dwm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dwm.c b/dwm.c
index 4465af1..1206b15 100644
--- a/dwm.c
+++ b/dwm.c
@@ -783,6 +783,7 @@ expose(XEvent *e)
 void
 focus(Client *c)
 {
+	Client *at;
 	if (!c || !ISVISIBLE(c))
 		for (c = selmon->stack; c && !ISVISIBLE(c); c = c->snext);
 	if (selmon->sel && selmon->sel != c)
@@ -802,6 +803,9 @@ focus(Client *c)
 		XDeleteProperty(dpy, root, netatom[NetActiveWindow]);
 	}
 	selmon->sel = c;
+	for (at = selmon->clients; at; at = at->next)
+		if (at != c && at->isfullscreen && ISVISIBLE(at))
+			setfullscreen(at, 0);
 	drawbars();
 }
 
-- 
2.19.1

